### DISCUSSION TITLE
Linking all ViewModels to share the Updatable interface

### DISCUSSION THREAD

**Discussion by Yuhan-Ut** [CSC207-2022F-UofT/course-project-bill-gates#37] [Ideas]
I think we should link all these viewmodel so they share the same updatable interface
but should we use the abstract class or an interface?
I would start with abstract class, but plz let me know if you think interface is better or any idea about this

**Comment by Yuhan-Ut** [comment]
however, an abstract class in the adapter  file is little bit strange, plz help me out with this

**Comment by xgao28** [comment]
#36  
I believe this discussion originated from your comment on this pull request.
For the updatable interface, it is used for BillDemo, and I believe what @Yunshan-522 did was on the correct track.

To be more specific, Updatable is for our different methods of various view models in our use case, and we should mention our own view model and then override it in BillDemo.

**Reply by Yuhan-Ut** [reply]
already make a pull request about the abstract class, you may want to have a look about that!
please tell anything or any concern about it! #38

**Comment by scottctd** [comment]
A view model is a **data structure class**, and it shouldn't have or implement any interfaces. Your wording is ambiguous.

The problem we need to address is: **Is there any way to make the interfaces of various Views into an all-in-one class?**

Also, how can we deal with a lot of repetitive ViewModels? 

In short, we should have **one interface and one view model** for a view, and one anther **interface and view model** for another view if exists.

We can observe that only `BillUpdateUseCase` needs a presenter and stuff like that. `AlterEntryUseCase`, `DeleteEntryUseCase`, and ... are all use cases to mutate data in the database. After altering the database, deleting an entry, or doing any operations on the bill table, we can simply update the table view by calling `BillUpdateUseCase`.
